bitnami/redis/templates/master/serviceaccount.yaml

```
@@ -3,7 +3,7 @@ Copyright VMware, Inc.
 SPDX-License-Identifier: APACHE-2.0
 */}}
 
-{{- if and .Values.master.serviceAccount.create (eq .Values.architecture "replication") (not .Values.sentinel.enabled) }}
+{{- if and .Values.master.serviceAccount.create (or (not (eq .Values.architecture "replication")) (not .Values.sentinel.enabled)) }}
```

I think this logic can be simplified by removing `.Values.architecture` from the equation.

The current conditional will cause the following:
* arch: standalone, sentinel: false -> master-SA created
* arch: standalone, sentinel: true -> Mode not supported, but master-SA enabled
* arch: replication, sentinel: false -> master-SA NOT created
* arch: replication, sentinel: true -> master-SA created

```suggestion
{{- if and .Values.master.serviceAccount.create (not .Values.sentinel.enabled) }}
```

The proposed conditional will cause the following:
* arch: standalone, sentinel: false -> master-SA created
* arch: standalone, sentinel: true -> Mode not supported and master-SA NOT enabled
* arch: replication, sentinel: false -> master-SA created
* arch: replication, sentinel: true -> master-SA NOT created

---

The last two results are reversed, so the original logic gives:
> The current conditional will cause the following:
> 
> arch: standalone, sentinel: false -> master-SA created
> arch: standalone, sentinel: true -> Mode not supported, but master-SA enabled
> arch: replication, sentinel: false -> master-SA created
> arch: replication, sentinel: true -> master-SA NOT created

If the standalone/sentinel enabled combintation is a "don't care", then the suggested logic is also fine. Though maybe there should be a clear failure message or something then that it's not supported.

```
{{- if and (eq .Values.architecture "standalone") (.Values.sentinel.enabled) -}}
fail "A standalone redis architecture with sentinel is not supported"
{{- end -}}
```

If the idea is to just install it as configured (even though it's unsupported), then the original logic is the one to go for.

Any changes here should also be portrayed over here btw: https://github.com/bitnami/charts/blob/bc4b1dcd3ba113f5a1e75b6a65ba522acc241dc5/bitnami/redis/templates/master/application.yaml#L7

---

Probably we should create a helper for both master and replication so that there is no duplicated logic to keep in sync. First let's agree on what the logic inside should be :smile: